magento/magento2
# Refactored Sitemap ItemProvider Interface and classes for Plugin customizability

### Description (*)

After retrieving the sitemap data, sitemap items are generated immediately though an inline defined function in array_map(). This makes it very difficult to update each item by itself, based on customer specific use cases. There are two solutions for this issue:

1. Iterate an extra time through every item in the collection after the getCollection() method. This is very time consuming, and is one of the current solutions.
2. Create a preference of the desired ItemProvider class and rewrite the getItems function for custom functionality. This is one of the current solutions, but undesirable from a maintenance perspective.
3. Remove the parameters of the item creation from the inline function. This can then be extended with a plugin to alter the data even more when needed. This is nog possible right now, but this PR creates the possibility.

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. Apply changes to a clean Magento 2.4 (or 2.3) environment
2. Generate a new sitemap 
3. Create a new plugin for Magento\Sitemap\Model\ItemProvider\Product::prepareParams(), updating the url to add an arbitrary suffix.
4. Generate a new sitemap
5. Notice the suffix added in all product url's.

### Questions or comments


### Contribution checklist (*)
